core/src/impl/Kokkos_Core.cpp

```diff
@@ -704,6 +717,20 @@ void parse_environment_variables(InitArguments& arguments) {
           "KOKKOS_DISABLE_WARNINGS if both are set. Raised by "
           "Kokkos::initialize(int narg, char* argc[]).");
   }
+  char* env_tuneinternals_str = std::getenv("KOKKOS_TUNE_INTERNALS");
+  if (env_tuneinternals_str != nullptr) {
+    std::string env_str(env_tuneinternals_str);  // deep-copies string
+    for (char& c : env_str) {
+      c = toupper(c);
+    }
+    if ((env_str == "TRUE") || (env_str == "ON") || (env_str == "1"))
+      tune_internals = true;
+    else if (!tune_internals)
```

Comment:
I think this logic is wrong. You are entering the environment variable is set to false branch, which means if !tune_internals everything is dandy. Note that even if you fix this (else if (tune_internals) without the !) there is still an issue which is also present for disable warnigns (thinking through the logic made that clear to me): if the user explicitly sets disable_warnings in the command line arguments to false, and also sets the enviornment variable to true, it will accept the environment variable, because there is no way to distinguish between "not set" and "set to false" for the command line arguments. We probably want to fix that some time. But for now I would be ok to just remove the ! here and then this should be good.

Reply:
Yeah, you're right (also, somehow I missed this). Really good catch

Reply:
Addressed in 1cc5bb6